subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/DefaultTaskProperties.java

```diff
@@ -233,6 +236,12 @@ public void visitEmptyOutputFileProperty(String propertyName, boolean optional,
             taskPropertySpecs.add(new DefaultValidatingProperty(propertyName, value, optional, ValidationActions.NO_OP));
         }
 
+        @Override
+        public void visitServiceReference(String propertyName, boolean optional, Provider<? extends BuildService<?>> serviceProvider, @Nullable String serviceName) {
+            // Service reference declared via annotation, validate it
+            taskPropertySpecs.add(new DefaultValidatingProperty(propertyName, StaticValue.of(serviceProvider), optional, ValidationActions.NO_OP));
```

Comment:
Here we should reuse the original `PropertyValue` we get in the `PropertyVisitor` instead of creating a new `StaticValue`.

Reply:
Thanks, done!